web/src/stats/stats.ts

```diff
@@ -9,6 +9,42 @@ import {page_params} from "../page_params";
 
 Plotly.register([PlotlyBar, PlotlyPie]);
 
+type DateFormatter = (date: Date) => string;
+
+type AggregatedData<T> = {
+    dates: Date[];
+    values: T;
```

Comment:
`T[]` is more appropriate for this property.

Reply:
I believe `values` is maps. For example, `values = {everyone:[], me:[]}`, while the values of `values` are arrays; the `values` itself can't be unifiable with `T[]`